aws/resource_aws_neptune_subnet_group.go

```diff
@@ -0,0 +1,264 @@
+		// AWS is down casing the name provided, so we compare lower case versions
+		// of the names. We lower case both our name and their name in the check,
+		// incase they change that someday.
+		if strings.ToLower(d.Id()) == strings.ToLower(aws.StringValue(s.DBSubnetGroupName)) {
+			subnetGroup = subnetGroups[0]
+		}
+	}
+
+	if subnetGroup.DBSubnetGroupName == nil {
+		return fmt.Errorf("Unable to find Neptune Subnet Group: %#v", subnetGroups)
+	}
+
+	d.Set("name", subnetGroup.DBSubnetGroupName)
+	d.Set("description", subnetGroup.DBSubnetGroupDescription)
+
+	subnets := make([]string, 0, len(subnetGroup.Subnets))
+	for _, s := range subnetGroup.Subnets {
+		subnets = append(subnets, aws.StringValue(s.SubnetIdentifier))
+	}
+	d.Set("subnet_ids", subnets)
+
+	// list tags for resource
+	// set tags
+
+	//Amazon Neptune shares the format of Amazon RDS ARNs. Neptune ARNs contain rds and not neptune.
+	//https://docs.aws.amazon.com/neptune/latest/userguide/tagging.ARN.html
+	arn := arn.ARN{
+		Partition: meta.(*AWSClient).partition,
+		Service:   "rds",
+		Region:    meta.(*AWSClient).region,
+		AccountID: meta.(*AWSClient).accountid,
+		Resource:  fmt.Sprintf("subgrp:%s", d.Id()),
+	}.String()
+	d.Set("arn", arn)
+	resp, err := conn.ListTagsForResource(&neptune.ListTagsForResourceInput{
+		ResourceName: aws.String(arn),
+	})
+
+	if err != nil {
+		log.Printf("[DEBUG] Error retreiving tags for ARN: %s", arn)
+	}
+
+	var dt []*neptune.Tag
+	if len(resp.TagList) > 0 {
+		dt = resp.TagList
+	}
+	d.Set("tags", tagsToMapNeptune(dt))
+
+	return nil
+}
+
+func resourceAwsNeptuneSubnetGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).neptuneconn
+	if d.HasChange("subnet_ids") || d.HasChange("description") {
+		_, n := d.GetChange("subnet_ids")
+		if n == nil {
+			n = new(schema.Set)
+		}
+		ns := n.(*schema.Set)
+
+		var sIds []*string
+		for _, s := range ns.List() {
+			sIds = append(sIds, aws.String(s.(string)))
+		}
+
+		_, err := conn.ModifyDBSubnetGroup(&neptune.ModifyDBSubnetGroupInput{
+			DBSubnetGroupName:        aws.String(d.Id()),
+			DBSubnetGroupDescription: aws.String(d.Get("description").(string)),
+			SubnetIds:                sIds,
+		})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	//Amazon Neptune shares the format of Amazon RDS ARNs. Neptune ARNs contain rds and not neptune.
+	//https://docs.aws.amazon.com/neptune/latest/userguide/tagging.ARN.html
+	arn := arn.ARN{
+		Partition: meta.(*AWSClient).partition,
+		Service:   "rds",
+		Region:    meta.(*AWSClient).region,
+		AccountID: meta.(*AWSClient).accountid,
+		Resource:  fmt.Sprintf("subgrp:%s", d.Id()),
+	}.String()
+	if err := setTagsNeptune(conn, d, arn); err != nil {
+		return err
+	} else {
+		d.SetPartial("tags")
+	}
+
+	return resourceAwsNeptuneSubnetGroupRead(d, meta)
+}
+
+func resourceAwsNeptuneSubnetGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	stateConf := &resource.StateChangeConf{
```

Comment:
This logic seems extraneous -- nothing appears to be retrying or setting a `pending` state. Seems like we can just call this normally:

```go
conn := meta.(*AWSClient).neptuneconn

input := neptune.DeleteDBSubnetGroupInput{
  DBSubnetGroupName: aws.String(d.Id()),
}

log.Printf("[DEBUG] Deleting Neptune Subnet Group: %s", d.Id())
_, err := conn.DeleteDBSubnetGroup(&input)
if err != nil {
  if isAWSErr(err, neptune.ErrCodeDBSubnetGroupNotFoundFault, "") {
    return nil
  }
  return fmt.Errorf("error deleting Neptune Subnet Group (%s): %s", d.Id(), err)
}

return nil
```